packages/adblocker/src/resources.ts

```diff
@@ -139,11 +202,77 @@ export default class Resources {
     return { body, contentType, dataUrl };
   }
 
+  public getScriptlet(name: string): string | undefined {
+    if (name.endsWith('.fn')) {
+      return undefined;
+    }
+
+    const scriptlet = this.scriptletsByName.get(name) || this.scriptletsByName.get(name + '.js');
+
+    if (scriptlet === undefined) {
+      this.scriptletsCache.set(name, '');
+      return undefined;
+    }
+
+    const [scriptletName] = scriptlet.names;
+    let script = this.scriptletsCache.get(scriptletName);
+
+    if (script !== undefined) {
+      if (script.length === 0) {
+        return undefined;
+      }
+      return script;
+    }
+
+    const dependencies = this.getScriptletDepenencies(scriptlet);
+    script = assembleScript(scriptlet.body, dependencies);
+
+    this.scriptletsCache.set(scriptletName, script);
+    return script;
+  }
+
+  private getScriptletDepenencies(scriptlet: Scriptlet): string[] {
+    const dependencies: Map<string, string> = new Map();
+    const queue: string[] = [...scriptlet.dependencies];
+
+    while (queue.length > 0) {
+      const dependencyName = queue.pop()!;
+      if (dependencies.has(dependencyName)) {
+        continue;
+      }
+      const dependency = this.scriptletsByName.get(dependencyName);
+      if (dependency === undefined) {
```

Comment:
Do we have static checks when building `resources.json` to make sure this never happens? (i.e. a dependency does not exist). Otherwise, maybe we should validate that in `parse(...)` (or both)

Reply:
We don't have a check for that including the backend but we assume the source of scriptlet (uBO) is correct.